lib/nuntiux/mocker.ex

```diff
@@ -308,57 +318,75 @@ defmodule Nuntiux.Mocker do
     :ok
   end
 
-  @spec maybe_run_expects(message, expects) :: expects_ran?
+  @spec maybe_run_expects(message, expects) :: expects_matched
         when message: term(),
              expects: expects(),
-             expects_ran?: boolean()
+             expects_matched: expects_matched()
   defp maybe_run_expects(message, expects) do
     Enum.reduce(
       expects,
-      _expects_ran = false,
+      @label_nomatch,
       fn
-        {_expect_id, _expect_fun}, true = _done ->
-          true
+        {_expect_id, _expect_fun}, {@label_match, _matched} = result ->
+          {:halt, result}
```

Comment:
I wanted to backport this specific bit to `nuntius`, but I'm not sure I'll be able to without writing a whole recursive function for it.

Reply:
I understand what it does, but… not having `reduce_while` in Erlang, it would be unnatural to just introduce that function to do this. I think `nuntius` is fine as it is.

Reply:
Sure. Maybe I can bring it up with erlang/otp directly (?) Thanks.